### ISSUE TITLE
Form: Fields do not indicate which ones are compulsory

### ISSUE THREAD

**Issue by inesdgomes** [international-labour-organization/designsystem#644]
Designs needs to be revised so form labels clearly show whether they're compulsory or not. The solution may be to add a red asterisk to the compulsory fields.

**Comment by justintemps** [comment]
@inesdgomes I'm moving this back to next sprint as I don't think we'll have time to work on it this week.

**Comment by inesdgomes** [comment]
@beatrizmartinmartins One small issue to tackle here. Basically we need a way to indicate which form fields are compulsory. The solution can simply be a red asterisk, if you think that works, but the designs need to be updated accordingly.

**Comment by beatrizmartinmartins** [comment]
Hey! So from what I am reading, it is better to indicate the ones that are optionals, so the required ones won't have any difference to the default inputs that we have now. For the optionals, I think adding a text saying "optional" is actually the best option because the red asterisk can be quite boring and usually on accessibility, we need to indicate to the user that the input field is mandatory or not, not only with color. So this is an idea I had and you can check it out on the Forms components page [https://www.figma.com/file/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?type=design&node-id=1433%3A3561&mode=design&t=bDY5mxoS52alNTX7-1](https://www.figma.com/file/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?type=design&node-id=1433%3A3561&mode=design&t=bDY5mxoS52alNTX7-1)

**Comment by inesdgomes** [comment]
Thanks @beatrizmartinmartins I understand the logic and agree the red asterisk doesn't sound like an accessible solution. However, at first sight I must say for me it's a bit counterintuitive as I would expect to be notified about the ones that are compulsory, rather than the ones that are optional. I'll also do some research and think this through. In the meantime, @justintemps any thoughts on this?

**Comment by inesdgomes** [comment]
Moving this to the backlog because forms are still not being used in the ILO website and we need to address other ticket first

**Comment by justintemps** [comment]
At this point I think users understand that asteriks in the form label means that the field is compulsory. If they're not sure, then the field will show an error message when they try to submit it. That seems like a suitably simple solution for now.

**Comment by inesdgomes** [comment]
@justintemps Agreed, from what I'm reading the asterisk is already a well-established convention. Let's use the asterisk for compulsory fields and add a note to the top of the form saying 'Asterisks indicate required fields'. @beatrizmartinmartins Could you please update the designs accordingly?

Also, @justintemps we can complement this solution with the 'aria-required' attribute. See here: https://www.tpgi.com/doing-whats-required-indicating-mandatory-fields-in-an-accessible-way/

_A more robust solution is to programmatically expose required form fields using the [aria-required](https://www.w3.org/TR/wai-aria-1.2/#aria-required) or [required](https://html.spec.whatwg.org/multipage/input.html#the-required-attribute) attribute. Doing so will result in screen readers explicitly announcing such form fields as “required”. This also provides a programmatic hook for Braille output, which can then abbreviate state information (e.g., “req” for “required”) to reduce verbosity and exploit the limited character space of a Braille display._

**Comment by beatrizmartinmartins** [comment]
@inesdgomes Removed the "optional" tag we had on forms, and included the asterisk. Is it okay if we make it red? I think it's more prominent and people see it clearly that that form field is mandatory. Please check [here](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=1433-3561&t=QYKSetKjFidmuRET-1)

**Comment by inesdgomes** [comment]
Thanks. Ready to be moved to development.
